src/core_plugins/timelion/server/series_functions/movingaverage.js

```diff
@@ -20,9 +24,21 @@ module.exports = new Chainable('movingaverage', {
   ],
   aliases: ['mvavg'],
   help: 'Calculate the moving average over a given window. Nice for smoothing noisey series',
-  fn: function movingaverageFn(args) {
+  fn: function movingaverageFn(args, tlConfig) {
     return alter(args, function (eachSeries, _window, _position) {
 
+      _window = (function () {
+        if (typeof _window === 'number') return _window;
+        // Ok, I guess its a datemath expression
+        const windowMilliseconds = toMS(_window);
+
+        // calculate how many buckets that _window represents
+        const intervalMilliseconds = toMS(tlConfig.time.interval);
+
+        // Round, floor, ceil? We're going with round because it splits the difference.
+        return Math.round(windowMilliseconds / intervalMilliseconds) || 1;
+      }());
```

Comment:
Consider using a simple if-statement to reassign the _window variable. A self-executing function like this is quite advanced usage and has little advantage now that we have block-scoping in JS.

e.g.
```
if (typeof_window !== 'number'){
    const windowMilliseconds = ....
    ...etc...
}
```